evcouplings/compare/sifts.py

```diff
@@ -137,8 +139,22 @@ def find_homologs_jackhmmer(**kwargs):
     if config["prefix"] is None:
         config["prefix"] = path.join(tempdir(), "compare")
 
+    check_required(
+        config,
+        [
+            "database", "prefix", "use_bitscores",
+            "domain_threshold", "sequence_threshold",
+            "nobias", "cpu",
+        ]
+    )
+
+    # run hmmsearch (possibly preceded by hmmbuild)
+    if search_pdb_by_alignment:
+        ar = hmmbuild_and_search(**config)
```

Comment:
here we have to make sure we pass the right alignment to use for searching - this has to be one of the raw files before introducing lowercase columns

Reply:
Right now I've implemented this as having the hmmbuild_and_search function call run_hmmbuild using the raw_focus_alignment_file key. I'm not sure whether it's better to call the function using the raw_alignment_file (ie the .sto formatted alignment). This could lead to a more 'deep' search but if we just end up throwing away the columns not found in our focus sequence anyway, not sure if there's a benefit. Plus, it seems like that would run the risk of more non-homologous structures. 

So two questions:
1. Is it sufficient just to explicitly use the correct key in hmmbuild_and_search, or should we manipulate the kwargs dictionary here to change the alignment_file key? 

2. Should we use the .sto file or the raw_focus_fasta file?

Reply:
We should use the alignment before it is turned into focus mode, so we get the insert and deletion probabilities in the HMM right. So the Stockholm alignment is the natural choice; also since it contains a lot of additional useful annotation.

Re the input alignment - I think the cleanest solution would be to name the input parameter of the search protocol "raw_alignment_file". That also signals to users of the function that we don't want the final alignment_file as an input here.